ext/intl/dateformat/dateformat_create.cpp

```diff
@@ -66,41 +69,46 @@ static int datefmt_ctor(INTERNAL_FUNCTION_PARAMETERS)
 
 	intl_error_reset(NULL);
 	object = return_value;
-	/* Parse parameters. */
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "s!ll|zzs",
-			&locale_str, &locale_len, &date_type, &time_type, &timezone_zv,
-			&calendar_zv, &pattern_str, &pattern_str_len) == FAILURE) {
-		return FAILURE;
-	}
+
+	ZEND_PARSE_PARAMETERS_START(3, 6)
+		Z_PARAM_STRING_OR_NULL(locale_str, locale_len)
+		Z_PARAM_LONG(date_type)
+		Z_PARAM_LONG(time_type)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_ZVAL(timezone_zv)
+		Z_PARAM_OBJ_OF_CLASS_OR_LONG_OR_NULL(calendar_obj, Calendar_ce_ptr, calendar_long, calendar_is_null)
+		Z_PARAM_STRING_OR_NULL(pattern_str, pattern_str_len)
+	ZEND_PARSE_PARAMETERS_END();
 
 	DATE_FORMAT_METHOD_FETCH_OBJECT_NO_CHECK;
 
 	if (DATE_FORMAT_OBJECT(dfo) != NULL) {
-		intl_errors_set(INTL_DATA_ERROR_P(dfo), U_ILLEGAL_ARGUMENT_ERROR,
-				"datefmt_create: cannot call constructor twice", 0);
-		return FAILURE;
+		intl_errors_set(INTL_DATA_ERROR_P(dfo), U_ILLEGAL_ARGUMENT_ERROR, "datefmt_create: cannot call constructor twice", 0);
+		zval_ptr_dtor(return_value);
+		RETURN_NULL();
```

Comment:
Why did this move into here? This made more sense to me in datefmt_create(), as it's only needed for that case. You can use ZEND_PARSE_PARAMETERS_END_EX to return FAILURE.

Reply:
The issue was that I couldn't return `FAILURE`, so `ZEND_PARSE_PARAMETERS_END_EX ` is very appreciated! :)